Timestream for LiveAnalytics Migration Tooling - Migration to Timestream for InfluxDB

fredjoonpark opened the pull request:
*Issue #, if available:*

*Description of changes:*

This PR adds READMEs and scripts for migrating to the Timestream for InfluxDB target.

The folder structure in `tools/python/liveanalytics_migration_scripts` now looks like the following:
```
├── README.md
├── requirements.txt
├── cardinality
│    ├── cardinality.py
│    └── README.md
├── targets
│    ├── rds_for_postgresql
│    └── timestream_for_influxdb
│        ├── ingestion/
│        ├── README.md
│        ├── transform/
│        └── validation/
└── unload
    ├── images
    ├── README.md
    ├── unload.py
    └── utils/
```


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fredjoonpark committed:
Initial structure for Timestream to InfluxDB migration

Refining structure for migration scripts

trevorbonas committed:
Add cardinality calculation script (#45)

Changes:
- The cardinality calculation script has been added.
    - `main.py` and `cardinality.py` have been combined into a single `cardinality.py` file.
    - A README has been added for the script.
- `timestream_utils.py` has been updated.
    - The class `timestreamUtility` has been renamed to `TimestreamUtility`.
    - Static functions for validating dimension, database, and table names have been added to `TimestreamUtility`.
    - The function `comma_separated_list` has been added to `TimestreamUtility`.
    - Broken `utils` imports in `timestream_utils.py` have been fixed.
    - `timestream_client` renamed to `timestream_write_client`, for clarity.
    - On line 105, the undefined variable `parition_count` has been corrected to `partition_count`.
    - `timestream_utils.py` has been reformatted with `ruff`.
- `s3_utils.py` has been updated.
    - Broken `utils` import has been fixed.
    - The class `s3Utility` has been changed to `S3Utility`.
- `unload.py` has been updated.
    - References to `s3Utility` and `timestreamUtility` have been updated.
    - Wording of various CLI options have been fixed.
    - All CLI options use `-` instead of `_` between words.
    - `datetime` object properly used.
    - Reformatted with `ruff`.

forestmvey committed:
Adding InfluxDB ingestion script (#46)

* Initial commit for integrating ingestion script.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Fixing ingestion_test naming for ingestion script.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Removing unecessary deprecation warning removal.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Fixing dead references to tests.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Update to using python3 for virtual environment.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Adding directories to the appropriate migration scripts in parent README.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Configuring docker to delete influx image after successful execution. Minor documentation revisions.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Adding test for resume functionality.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

---------

Signed-off-by: forestmvey <forestv@bitquilltech.com>

fredjoonpark committed:
Add migration validation script  (#47)

Adding validation script

trevorbonas committed:
Add transform script (#48)

Changes:
- `transform.py` has been added, allowing already-unloaded Timestream for LiveAnalytics data to be transformed into line protocol using Athena.
- `athena_utils.py` has been added.
- `TimestreamUtility` now has a wrapper function for `list_tables`.
- `S3Utility` now has the method `s3_bucket_exists` for checking the existence of an S3 bucket.
- The region for the `S3Utility` constructor defaults to `None`, causing the AWS configured default region to be used. An `S3Utility` object can now be created simply with `S3Utility()`.
- A single `requirements.txt` file has been added to `liveanalytics_migration_scripts/` and is intended to be used for all scripts within.
- All other `requirements.txt` files have been removed.
- The path where data would be unloaded by `unload.py` used to include a space for example,
`unload-2025-05-07 22:31:43`. This space has been replaced with `-`.

forestmvey committed:
Add LiveAnalytics migration readme and Timestream for InfluxDB target README (#49)

* Initial commit for migration README, starting timestream for InfluxDB migration README.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Initial version of the Timestream for InfluxDB migration README.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

* Minor revisions to workflow stages and updating script names and parameters.

Signed-off-by: forestmvey <forestv@bitquilltech.com>

---------

Signed-off-by: forestmvey <forestv@bitquilltech.com>

trevorbonas commented on tools/python/liveanalytics_migration_scripts/unload/unload.py:
```diff
@@ -44,9 +46,12 @@
+    end_time = args.end_time
+    migration_tag = (
+        args.migration_tag
+        or f"unload-{datetime.now(timezone.utc).strftime('%Y-%m-%d-%H:%M:%S')}"
```
This is intentional. The `unload` path was changed from
```
'unload-%Y-%m-%d %H:%M:%S'
```
to
```
'unload-%Y-%m-%d-%H:%M:%S'
```
This means users won't have to escape the space when syncing the bucket later.

fredjoonpark committed:
additional rebase cleanup

fredjoonpark committed:
Consolidating env, fixing links

trevorbonas committed:
Add time range to cardinality script (#52)

Changes:
- Add `--start-time` and `--end-time` to `cardinality.py`.
- Fix typos `validation/README.md`.

fredjoonpark committed:
Consolidating installation docs, minor cleanup (#54)

Consolidating installation docs, minor cleanup

fredjoonpark committed:
restructuring
